git: fix bb5 users sysconfig

tristan0x opened the pull request:
This pull-request allows me to use a recent version of git on BB5 instead of the one provided by the system. (1.8.3.1 has been released on **10-Jun-2013**)

I had to remove some system packages in `packages.yml` to make it work. I don't know if there is another way. See commit message for more details.

pramodk commented on sysconfig/bb5/users/packages.yaml:
```diff
@@ -127,10 +123,6 @@ packages:
-        paths:
-            pcre@8.32: /usr
-        version: [8.32]
     perl:
```
I assume you might have seen error like `==> Error: An unsatisfiable variant constraint has been detected for spec:`. This is because `git` ask for `depends_on('pcre+jit'...')` and our spec is not sufficient enough. Instead of removing package (which is installed on system), it's sufficient to do:

```
-            pcre@8.32: /usr
+            pcre@8.32+jit+utf: /usr
```

this I have just tested and it works fine.

tristan0x commented:
Ok. Let me refine our spec then.

Commit:
git package: fix bb5 users sysconfig

* allows `git` module
* remove system gettext to prevent link issue:
    ```
    32 errors found in build log:
         646        LINK git-http-push
         647        LINK git
         648        LINK git-credential-cache--daemon
         649        LINK git-upload-pack
         650        LINK git-credential-cache
         651        LINK git-remote-testsvn
      >> 652    ld: cannot find -lintl
    ```

tristan0x commented:
@pramodk I force pushed the commit

I have to remove `gettext` otherwise I have link errors:
```
32 errors found in build log:
     646        LINK git-http-push
     647        LINK git
     648        LINK git-credential-cache--daemon
     649        LINK git-upload-pack
     650        LINK git-credential-cache
     651        LINK git-remote-testsvn
  >> 652    ld: cannot find -lintl
     653    ld: skipping incompatible /usr/lib/librt.so when searching for -lrt
     654    ld: skipping incompatible /usr/lib/librt.so when searching for -lrt
     655    ld: skipping incompatible /usr/lib/libm.so when searching for -lm
     656    ld: skipping incompatible /usr/lib/libm.so when searching for -lm
```

Is that fine with you?

pramodk commented:
When I mentioned it works, I meant it for `pcre` only. Yes, we have to remove gettext.

tristan0x commented:
Thank you @pramodk !
